### PR TITLE
AP_Scripting: added lua binding get_last_update for efi and esc

### PR DESCRIPTION
**Pull request by loki077**
We need this data available for Lua scripts

### CODE REVIEW

**Review by magicrub — APPROVED** [review]

File: libraries/AP_Scripting/generator/description/bindings.desc [hunk 1]
```diff
@@ -441,6 +441,7 @@ singleton AP_ESC_Telem method get_usage_seconds boolean uint8_t 0 NUM_SERVO_CHAN
 singleton AP_ESC_Telem method update_rpm void uint8_t 0 NUM_SERVO_CHANNELS uint16_t'skip_check float'skip_check
 singleton AP_ESC_Telem method update_telem_data void uint8_t 0 NUM_SERVO_CHANNELS AP_ESC_Telem_Backend::TelemetryData uint16_t'skip_check
 singleton AP_ESC_Telem method set_rpm_scale void uint8_t 0 NUM_SERVO_CHANNELS float'skip_check
+singleton AP_ESC_Telem method get_last_telem_data_ms uint32_t uint8_t 0 NUM_SERVO_CHANNELS
```
Comment (peterbarker) [inline libraries/AP_Scripting/generator/description/bindings.desc:444]:
```suggestion
singleton AP_ESC_Telem method get_last_telem_data_ms uint32_t uint8_t 0 ESC_TELEM_MAX_ESCS
```

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_Scripting/generator/description/bindings.desc (hunk 1 above)
Comment (loki077) [reply libraries/AP_Scripting/generator/description/bindings.desc:444]: In this case shouldn't all the other ESC_telem defines be change from NUM_SERVO_CHANNELS to ESC_TELEM_MAX_ESCS

Comment (peterbarker) [reply libraries/AP_Scripting/generator/description/bindings.desc]: > In this case shouldn't all the other ESC_telem defines be change from NUM_SERVO_CHANNELS to ESC_TELEM_MAX_ESCS

Possibly?  I based this suggestion on the sanity check in the top of that method, nothing more.

**Comment by loki077** [comment]
@peterbarker @tridge I have made the requested changes.

File: libraries/AP_Scripting/generator/description/bindings.desc [hunk 2]
```diff
@@ -438,16 +438,17 @@ userdata AP_ESC_Telem_Backend::TelemetryData field motor_temp_cdeg int16_t'skip_
...
-singleton AP_ESC_Telem method update_rpm void uint8_t 0 NUM_SERVO_CHANNELS uint16_t'skip_check float'skip_check
-singleton AP_ESC_Telem method update_telem_data void uint8_t 0 NUM_SERVO_CHANNELS AP_ESC_Telem_Backend::TelemetryData uint16_t'skip_check
-singleton AP_ESC_Telem method set_rpm_scale void uint8_t 0 NUM_SERVO_CHANNELS float'skip_check
+singleton AP_ESC_Telem method get_rpm boolean uint8_t 0 ESC_TELEM_MAX_ESCS float'Null
+singleton AP_ESC_Telem method get_temperature boolean uint8_t 0 ESC_TELEM_MAX_ESCS int16_t'Null
+singleton AP_ESC_Telem method get_motor_temperature boolean uint8_t 0 ESC_TELEM_MAX_ESCS int16_t'Null
+singleton AP_ESC_Telem method get_current boolean uint8_t 0 ESC_TELEM_MAX_ESCS float'Null
+singleton AP_ESC_Telem method get_voltage boolean uint8_t 0 ESC_TELEM_MAX_ESCS float'Null
+singleton AP_ESC_Telem method get_consumption_mah boolean uint8_t 0 ESC_TELEM_MAX_ESCS float'Null
+singleton AP_ESC_Telem method get_usage_seconds boolean uint8_t 0 ESC_TELEM_MAX_ESCS uint32_t'Null
```
Comment (IamPete1) [inline libraries/AP_Scripting/generator/description/bindings.desc:447]: I think we can just `'skip_check` for all these that return bool. The result will be null and `AP_ESC_Telem` should deal with it correctly. The problem with using a define is that there is no way for the script to get the value.

Comment (Lokesh-Carbonix) [reply libraries/AP_Scripting/generator/description/bindings.desc:447]: Do you want all ESC_TELEM_MAX_ESCS to be replaced with 'skip_check. I am not sure of the define implementation and have been following the previous implementation. Not sure if it wil brake anything else. @IamPete1 if you are good with it I will update the PR.

Comment (loki077) [reply libraries/AP_Scripting/generator/description/bindings.desc:447]: @robertlong13 FYI

Comment (IamPete1) [reply libraries/AP_Scripting/generator/description/bindings.desc:447]: Yeah, we should be good to remove them all.

**Review by tridge — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged it, thanks!

**Pull request merged**